packages/sdk/src/sdk.ts

```diff
@@ -477,7 +477,8 @@ export class NxtpSdk {
         }
 
         // check if the price changes unfovorably by more than the slippage tolerance(percentage).
-        const priceImpact = ((parseFloat(data.bid.amountReceived) - parseFloat(amount)) / parseFloat(amount)) * 100;
+        const priceImpact =
+          (parseFloat(BigNumber.from(data.bid.amountReceived).sub(amount).toString()) / parseFloat(amount)) * 100;
```

Comment:
You can't parseFloat on BigNumbers. You need to do integer division. We have the `calculateExchange` function in the utils that's meant for this.

Reply:
We need float division cause price impact is going to be in percentage.
Not sure how we can use the `calculateExchange()` func.